
<launch>
  <env name="PYTHONPATH" value="$(env PYTHONPATH)" />

  <arg name="nao_ip" default="$(optenv NAO_IP 127.0.0.1)" />
  <arg name="nao_port" default="$(optenv NAO_PORT 9559)" />
  <arg name="resolution" default="2" />
  <arg name="color_space" default="13" />
  <arg name="frame_rate" default="15" />
  <node pkg="nao_sensors" type="camera.py" name="nao_camera" required="true" args="--pip=$(arg nao_ip) --pport=$(arg nao_port)" output="screen">
     <param name="calibration_file_bottom" type="string" value="file://$(find nao_sensors)/share/nao_camera_bottom_640x480.yaml"/>
     <param name="calibration_file_top" type="string" value="file://$(find nao_sensors)/share/nao_camera_top_640x480.yaml"/>
     <param name="resolution" type="int" value="$(arg resolution)" /> <!--QQVGA=0, QVGA=1, VGA=2 -->
     <param name="color_space" type="int" value="$(arg color_space)" /> <!-- 9 = YUV422, 11 = RGB, 13 = BGR -->
     <param name="frame_rate" type="int" value="$(arg frame_rate)" />
     <param name="camera_switch" type="int" value="0" /> --> <!-- 0 = top, 1 = bottom -->
  </node>

</launch>
